packages/repository/test/unit/decorator/relation.decorator.unit.ts

```diff
@@ -124,3 +124,43 @@ describe('relation decorator', () => {
     });
   });
 });
+
+describe('getModelRelations', () => {
+  it('returns relation metadata for own and inherited properties', () => {
+    @model()
+    class AccessToken extends Entity {
+      @property({id: true})
+      userId: number;
+    }
+
+    @model()
+    class User extends Entity {
+      @hasMany(AccessToken)
+      accessTokens: AccessToken[];
+    }
+
+    @model()
+    class Order extends Entity {
+      @property({id: true})
+      customerId: number;
+    }
+
+    @model()
+    class Customer extends User {
+      @hasMany(Order)
+      orders: Order[];
+    }
+
+    const relations = getModelRelations(Customer);
+    expect(relations).to.deepEqual({
+      accessTokens: {
+        keyTo: 'userId',
```

Comment:
wouldn't it be `customerId`?
Otherwise when we have multiple models extend `User`, they will share the same fk which is `userId`...

Reply:
No, it's `userId` because Customer inherited the relation the User class.

In LB 3.x, we call this a "polymorphic relation" and add another property to `accessToken` to distinguish between different targets that can have the same `userId`. For example, `accessToken` can have two properties: `userId` for the id (PK/FK) value, `userModel` for a name of the target model class (User, Customer, etc.)

The purpose of the test is just to verify that the new function returns all relations - defined on the model and inherited from parents.

Reply:
> we call this a "polymorphic relation"
oh ok I got it 👍 it's polymorphic.